### PR TITLE
Rhel 10 beta (HMS-5011)

### PR DESCRIPTION
**Pull request by croissanne**
Adds rhel 10 beta support :)

### CODE REVIEW

**Comment by regexowl** [comment]
Changes from `npm run api` will need to be added as well I think.

File: src/Components/CreateImageWizard/utilities/requestMapper.ts [hunk 1]
```diff
@@ -133,7 +134,11 @@ const convertFilesystemToPartition = (filesystem: Filesystem): Partition => {
  * @param distribution blueprint distribution
  */
 const getLatestRelease = (distribution: Distributions) => {
-  return distribution === RHEL_9_BETA
+  return distribution === RHEL_10_BETA
+    ? (RHEL_10_BETA as Distributions)
+    : distribution.startswith('rhel-10')
+    ? RHEL_10
```
Comment (regexowl) [inline src/Components/CreateImageWizard/utilities/requestMapper.ts:140]: There's no RHEL 10 option in API, should there be?

File: src/Components/CreateImageWizard/steps/Oscap/Oscap.tsx [hunk 2]
```diff
@@ -460,6 +465,8 @@ const ComplianceSelectOption = ({ policy }: ComplianceSelectOptionPropType) => {
 // so just use the ones from the major release.
 export const removeBetaFromRelease = (dist: Distributions): Distributions => {
   switch (dist) {
+    case RHEL_10_BETA:
+      return RHEL_10;
```
Comment (regexowl) [inline src/Components/CreateImageWizard/steps/Oscap/Oscap.tsx:469]: Same here, no RHEL_10 in Distributions.

Comment (croissanne) [reply src/Components/CreateImageWizard/steps/Oscap/Oscap.tsx:469]: ugh right, hmm how's that going to work :confused:

File: src/Components/CreateImageWizard/utilities/requestMapper.ts (hunk 1 above)
Comment (croissanne) [reply src/Components/CreateImageWizard/utilities/requestMapper.ts:140]: right I see what you mean, hmmm

File: src/Components/CreateImageWizard/steps/Oscap/Oscap.tsx (hunk 2 above)
Comment (croissanne) [reply src/Components/CreateImageWizard/steps/Oscap/Oscap.tsx:469]: actually I think we can just return the constant and cast it?
bit hacky but should work until rhel 10 actually exists.

**Comment by regexowl** [comment]
And `ImagesTable/Release.tsx` will need `'rhel-10-beta': 'RHEL 10 Beta',` added as well.

**Comment by regexowl** [comment]
Hmm, repo filtering will need and update for RHEL 10

image

Oscap is probably also confused by the RHEL 10

image

**Comment by regexowl** [comment]
Tried to create a blueprint

image

File: src/Components/CreateImageWizard/steps/ImageOutput/ReleaseSelect.tsx [hunk 3]
```diff
@@ -45,7 +46,7 @@ const ReleaseSelect = () => {
   };
 
   const setDescription = (key: Distributions) => {
-    if (key === RHEL_9_BETA) {
+    if (key === RHEL_9_BETA || key == RHEL_10_BETA) {
```
Comment (regexowl) [inline src/Components/CreateImageWizard/steps/ImageOutput/ReleaseSelect.tsx:49]:
```suggestion
    if (key === RHEL_9_BETA || key === RHEL_10_BETA) {
```

File: src/Components/ImagesTable/Release.tsx [hunk 4]
```diff
@@ -26,6 +26,7 @@ const Release = ({ release }: ReleaseProps) => {
     'rhel-93': 'RHEL 9.3',
     'rhel-94': 'RHEL 9.4',
     'rhel-10-nightly': 'RHEL 10',
+    'rhel-10-beta': 'RHEL 10 Beta',
```
Comment (avitova) [inline src/Components/ImagesTable/Release.tsx:29]: Amazing, I just ran into this problem after updating the image builder API. Thanks for addressing this!:)

**Review by regexowl — COMMENTED** [review]
Looks good, the only problem is the custom repos not being filtered by release for RHEL 10. 

OpenSCAP doesn't currently work for RHEL 10, but it might not be a blocker as it's just a "missing" cutomization option that could be added in a follow up.

**Comment by regexowl** [comment]
We could also probably hide the repos step for RHEL 10 Beta or add some kind of alert that it's "under construction" if this needs to go out quickly.

**Comment by croissanne** [comment]
konflux fails because rhsm is down.

**Pull request merged**